### ISSUE TITLE
Add "Finish Assessment" button for all assessments

### ISSUE THREAD

**Issue by firasm** [PrairieLearn/PrairieLearn#9261]
TIL (thanks to Jessica Wong for asking and @nwalters512 for answering) that the "Finish Assessment" button only appears depending on your access rule settings:

> My understanding is that that button only appears if real-time grading is disabled or if `showClosedAssessment: false` is set.

<img width="1119" alt="Screenshot 2024-01-19 at 4 18 53 PM" src="https://github.com/PrairieLearn/PrairieLearn/assets/2507459/a7ccc5b4-0c56-4b4c-96d7-40b4b961583b">

I think that "finish assessment" button should appear all the time rather than the standard workflow of this message:

> When you are done, please logout and close your browser; there is no need to do anything else. If you have any saved answers when you leave, they will be automatically graded before your final score is computed.

<img width="1207" alt="Screenshot 2024-01-19 at 4 24 21 PM" src="https://github.com/PrairieLearn/PrairieLearn/assets/2507459/fc95e56a-9ed0-4c60-8904-081d533b9782">

--------------

I understand that the button doesn't actually "do" anything, but honestly, the satisfaction of clicking a button to end something is worth it compared to the anxiety of ending an exam by closing a browser window and walking away. I think for other cases, adding that button there will smooth out the workflow, we can even leave that text about saved answers there as-is.

Jessica Wong also said:

> One of our first year programming classes (for non-majors) typically has a population that needs more reassurance over things like this. I think it would help them psychologically but I can always do the “say it 10 times in class and hope someone heard” thing :)

**Comment by jonatanschroeder** [comment]
I believe it's based on assessment type, not access rules.

**Comment by nwalters512** [comment]
While it's true that it's only ever shown for Exam-type assessments, it's not shown for all Exams. The actual conditions for when its shown is based on multiple factors, including access rules:

https://github.com/PrairieLearn/PrairieLearn/blob/88ee7bd271142a5a02c3cf12c39d1cf070651a52/apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.ejs#L310-L347

**Comment by mwest1066** [comment]
I'm not sure this is a good idea. Once you click it, there is no way to undo the click. While I think the risk of clicking this accidentally is low, I'm hesitant to add a button that has no purpose and can only cause someone to lock themselves out of further attempts.

**Comment by firasm** [comment]
Fair enough - I agree it makes less sense for HW types, but maybe can discuss it some more - the fact it doesn't appear consistently for all Exam types is a bit frustrating.

This is somewhat complicated by the future merging of HW and Exam type assessments. 

I guess it would be nice if assessments could be explicitly closed.

**Comment by mwest1066** [comment]
@firasm and I discussed this today and he pointed out that a key use case for the "Finish assessment" button is for students to indicate that their work is ready for manual grading. It would be better to entirely remove "Finish assessment" and replace it with a "Submit for grading" button, which would apply when real time grading is false or there are manually graded questions. This will be part of the implementation of alternative grading.
